File: lib/forecast-parser.js
const ThreeHourForecast = require('./three-hour-forecast');

function ForecastParser(list) {
  this.list = list.map((item) => {
    return new ThreeHourForecast(item);
  }).filter((item) => item.today());
}

ForecastParser.prototype.rainPrediction = function() {
  if (this.list.some((item) => item.rain())) {
    return "It's going to rain";
  } else {
    return 'Nope';
  }
}

module.exports = ForecastParser;

File: forecast_parser.rb
# frozen_string_literal: true

require_relative 'three_hour_forecast'

class ForecastParser
  def initialize(list)
    @list = list.map { |item| ThreeHourForecast.new(item) }.select(&:today?)
  end

  def rain_prediction
    if @list.any?(&:rain?)
      'Yes, it will!'
    else
      'Nope'
    end
  end
end
